### ISSUE TITLE
[Enhance] Add interpolation mode argument to transformations

### ISSUE THREAD

**Issue by abhinavdhere** [kornia/kornia#811]
## 🚀 Feature

Transforms such as Translate in geometry.transform do not have an option to set interpolation type, whereas the underlying warp_affine which they use do have the option. Rotate already has this option, which is not mentioned in the docs. Adding this to all transforms that require interpolation may be a useful addition.

This is a similar request to #218 which requested for setting padding mode. In many use cases, it may be useful to change the interpolation type when doing transforms. One such use case is when applying geometric transforms to segmentation masks where the values must remain limited to certain unique integer values. Having nearest-neighbor interpolation would preserve these values, whereas the default bilinear would not.
## Pitch

Add an argument to set interpolation mode in transformations of geometry.transform. This argument can be carried forward to warp_affine function called by these transform functions.

**Comment by shijianjian** [comment]
Agreed. Are you okay to go for a PR to add this feature?

**Comment by abhinavdhere** [comment]
Yes, I can work on a PR.

**Comment by edgarriba** [comment]
@pmeier is this related to https://github.com/kornia/kornia/pull/806 ?

**Comment by pmeier** [comment]
> is this related to #806 ?

Not directly. If the change proposed here is happening, #806 needs to be expanded to cover this as well. Basically, #806 should be applied everywhere we internally use `torch.nn.functional.interpolate` **and** do not have a fixed interpolation mode.

**Comment by abhinavdhere** [comment]
I'm busy with a work deadline these days. When I get some time, I'll populate a list of changes required for this change, and if you find it okay I'll send a PR. @edgarriba 
It would be my first PR on this project, so I'm excited to contribute.

**Comment by edgarriba** [comment]
@abhinavdhere this sounds awesome :)
